src/app/shell/result/result.component.ts

```diff
@@ -58,23 +58,36 @@ export class ResultComponent implements OnInit, OnDestroy {
   constructor(
     private actions$: Actions,
     private store: Store,
-    public navigationBarService: NavigationBarService
+    public navigationBarService: NavigationBarService,
+    private route: ActivatedRoute,
+    private router: Router
   ) { }
 
   ngOnInit(): void {
 
-    this.store.dispatch(
+    this.route.params
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((params: Params) => {
+      this.currentView = (params.param === this.viewType.map) ? this.viewType.map : this.viewType.data;
```

Comment:
probably you can avoid this comparing. Just to   this.currentView = params.param ? I believe param is equal to viewType value